angular-ngrx-scss/src/app/shared/components/repo-controls/repo-controls.component.html

```diff
@@ -7,39 +7,37 @@
       placeholder="Find a repository..."
       [formControl]="searchInput"
     />
-    <button class="control-dropdown" role="button">
-      <span class="control-dropdown__name">Type</span>
-      <span
-        class="control-dropdown__caret"
-        appOcticon="chevron-down"
-        size="14"
-      ></span>
-    </button>
-    <button class="control-dropdown" role="button">
-      <span class="control-dropdown__name">Language</span>
-      <span
-        class="control-dropdown__caret"
-        appOcticon="chevron-down"
-        size="14"
-      ></span>
-    </button>
-    <button class="control-dropdown" role="button">
-      <span class="control-dropdown__name">Sort</span>
-      <span
-        class="control-dropdown__caret"
-        appOcticon="chevron-down"
-        size="14"
-      ></span>
-    </button>
+    <app-repos-filter-dropdown
+      name="Type"
+      description="Select type"
+      [items]="typeFilters"
+      [current]="selectFilterByType$ | async"
+      (setFilter)="handleTypeClick($event)"
+    ></app-repos-filter-dropdown>
+    <app-repos-filter-dropdown
+      name="Language"
+      description="Select type"
+      [items]="typeFilters"
+      (setFilter)="handleTypeClick($event)"
+    ></app-repos-filter-dropdown>
+    <app-repos-filter-dropdown
+      name="Sort"
+      description="Select type"
+      [items]="typeFilters"
+      (setFilter)="handleTypeClick($event)"
+    ></app-repos-filter-dropdown>
```

Comment:
I assume these are temporary just for styling.

blocker(code-reuse): Right now the styles for the dropdown only live in this ad-hoc component. We will need to re-use them for the other dropdowns in the future as well. I think it would make more sense to create a re-usable dropdown component that lets us pass options in via content projection instead. We could then use that component in all 3 cases here and pass in the relevant filters as children elements.

Reply:
yes, there are just temporary for styling, in the consecutive PRs this will change to the proper name, description, etc.

Agree with you we should re-use a component and passing items via content project looks a great solution, but this is not part of this ticket scope, maybe we need to create another ticket. Just add this component because didn't have anything to complete the functionality described on the AC.
What do you think ? @Reshurum

Reply:
Since this component will be added to the codebase either way, I'd much rather see the reusable component made in this PR and used instead of the temporary one that would then need to be removed and/or recreated later. If we're going to add it either way, let's go ahead and add it the right way.

- [ ] Please go ahead and move this to a shared reusable component